src/elevenlabs/conversational_ai/conversation.py

```diff
@@ -0,0 +1,210 @@
+from abc import ABC, abstractmethod
+import base64
+import json
+import threading
+from typing import Callable, Optional
+
+from websockets.sync.client import connect
+
+from ..base_client import BaseElevenLabs
+
+
+class AudioInterface(ABC):
+    """AudioInterface provides an abstraction for handling audio input and output."""
+
+    @abstractmethod
+    def start(self, input_callback: Callable[[bytes], None]):
+        """Starts the audio interface.
+
+        Called one time before the conversation starts.
+        The `input_callback` should be called regularly with input audio chunks from
+        the user. The audio should be in 16-bit PCM mono format at 16kHz. Recommended
+        chunk size is 4000 samples (250 milliseconds).
+        """
+        pass
+
+    @abstractmethod
+    def stop(self):
+        """Stops the audio interface.
+
+        Called one time after the conversation ends. Should clean up any resources
+        used by the audio interface and stop any audio streams. Do not call the
+        `input_callback` from `start` after this method is called.
+        """
+        pass
+
+    @abstractmethod
+    def output(self, audio: bytes):
+        """Output audio to the user.
+
+        The `audio` input is in 16-bit PCM mono format at 16kHz. Implementations can
+        choose to do additional buffering. This method should return quickly and not
+        block the calling thread.
+        """
+        pass
+
+    @abstractmethod
+    def interrupt(self):
+        """Interruption signal to stop any audio output.
+
+        User has interrupted the agent and all previosly buffered audio output should
+        be stopped.
+        """
+        pass
+
+
+class Conversation:
+    client: BaseElevenLabs
+    agent_id: str
+    requires_auth: bool
+
+    audio_interface: AudioInterface
+    callback_agent_response: Optional[Callable[[str], None]]
+    callback_user_transcript: Optional[Callable[[str], None]]
+    callback_latency_measurement: Optional[Callable[[int], None]]
```

Comment:
callback_agent_response_correction missing here?

Reply:
Oops, thanks for catching